src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -0,0 +1,172 @@
+            }
+        }
+
+        public override bool CanSeek
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override bool CanWrite
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override long Length
+        {
+            get
+            {
+                CheckDisposed();
+                throw new NotSupportedException();
+            }
+        }
+
+        public override long Position
+        {
+            get
+            {
+                CheckDisposed();
+                throw new NotSupportedException();
+            }
+
+            set
+            {
+                CheckDisposed();
+                throw new NotSupportedException();
+            }
+        }
+
+        public override void Flush()
+        {
+            // Nothing to do.
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException("buffer");
+            }
+
+            if (offset < 0)
+            {
+                throw new ArgumentOutOfRangeException("offset");
+            }
+
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException("count");
+            }
+
+            if ((long)offset + (long)count > (long)buffer.Length)
+            {
+                throw new ArgumentException("buffer");
+            }
+
+            CheckDisposed();
+
+            uint bytesRead = 0;
+            var pinnedHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
+            var result = Interop.WinHttp.WinHttpReadData(
+                _requestHandle,
+                Marshal.UnsafeAddrOfPinnedArrayElement(buffer, offset),
+                (uint)count,
+                out bytesRead);
+            pinnedHandle.Free();
+            if (!result)
+            {
+                throw new IOException(SR.net_http_io_read, WinHttpException.CreateExceptionUsingLastError());
+            }
+
+            return (int)bytesRead;
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            CheckDisposed();
+            throw new NotSupportedException();
+        }
+
+        public override void SetLength(long value)
+        {
+            CheckDisposed();
+            throw new NotSupportedException();
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            CheckDisposed();
+            throw new NotSupportedException();
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && !_disposed)
+            {
+                _disposed = true;
+
+                _requestHandle.DangerousRelease();
+                _connectHandle.DangerousRelease();
+                _sessionHandle.DangerousRelease();
+
```

Comment:
Another place where the object being GC'd without `Dispose` will leak handles (and associated data).  Unless I'm horribly mistaken the AddRef/Release calls shouldn't be needed, because that's what the SafeHandle is already doing for free.